### DISCUSSION TITLE
Possibility to manage Groups by templates

### DISCUSSION THREAD

**Discussion by NT3R117** [ebullient/obsidian-task-collector#355] [Ideas]
Hello,

First of all thanks for this great feature, it helps a lot.

I would like to suggest an additional feature to manage groups by templates or add templates to specifics groups.

My problem is that I have 3 types of check lists and they don't actually need the same collections areas, I know i can use differents Groups but it will be great to use the same marks with different behaviours depending on the template we use.

Thanks for your time 🥇

**Comment by ebullient** [comment]
Can you give me an example of what you mean by a template?

**Reply by NT3R117** [reply]
Hello,

Of course, 

By template I mean a note template file located into a template folder which is managed by the Core Plugin "Templates".

If something into the new note file indicates that we used a specific template, or if something inside the note's body or note's properties match a specific word or tag, then a specific group of your plugin can be used to manage task (checkboxes) of the note.

![Screenshot_20231129_230223_Obsidian](https://github.com/ebullient/obsidian-task-collector/assets/61501206/14ee7de9-7a76-41c9-abe0-2d86f70e68da)
![Screenshot_20231129_230244_Obsidian](https://github.com/ebullient/obsidian-task-collector/assets/61501206/f7bed285-2591-4844-bf00-839edfd81bfe)

Thanks for your time 😀

**Reply by ebullient** [reply]
Yea.. that was my worry. I have no association between the note and the template used to create it.

You have to think about the note as something standalone.

What *could* be possible is a tag or metadata property, i.e. configure a group to be used only if a certain tag or metadata field is defined in the frontmatter (which could be assigned by the template)

**Reply by NT3R117** [reply]
Indeed, there is no reason to keep this information in the new note I guess...

As you said, if it could be possible to associate these elements to groups, it will be a great feature in my opinion : 
- A regex to search for words within the note and identify it.
- One or more tags (this will make groups usables for multiple templates / purposes ),
- Some other metadata or property.

Thanks for your feedback 😀

**Comment by ebullient** [comment]
Not sure I'm going to do this one. The existing marking groups allow a lot of customization, and tying those to another vector will make things both harder to maintain and harder to understand (from a config perspective).

**Reply by NT3R117** [reply]
Hello,

Ok, I understand, thanks for your time :)
